solvers/mathematical_program_result.h

```diff
@@ -218,6 +231,51 @@ class MathematicalProgramResult final {
     return value;
   }
 
+  /**
+   * Gets the dual solution associated with a constraint.
+   *
+   * We interpret the dual variable value as the "shadow price" of the original
```

Comment:
I like it! I think it is the cleanest solution. I tried to do some pen and paper tests to see if we can run into some contradictions sticking with this interpretation in all the cases, but I could not find any.